### PR TITLE
fixed multiple errors in the LFOM code. 

### PR DESCRIPTION
**Pull request by monroews**
Confirmed that it produces reasonable designs over a range of flow rates.
Fixed multiple unit errors
Fixed algorithm so that it minimizes error

Added all inputs as keyword inputs.

### CODE REVIEW

File: aguaclara/design/lfom.py [hunk 1]
```diff
@@ -114,14 +111,14 @@ def n_orifices_per_row_max(self):
...
-        b = self.orifice_diameter + self.s_orfice
+        b = self.orifice_diameter + self.s_orifice
 
         return math.floor(c/b)
 
     @property
     def flow_ramp(self):
         """An equally spaced array representing flow at each row."""
-        return np.linspace(self.q / self.n_rows, self.q, self.n_rows)*self.q.units
+        return np.linspace(1 / self.n_rows, 1, self.n_rows)*self.q
```
Comment (eak24) [inline aguaclara/design/lfom.py:121]: This is a lot cleaner - I didn't realize np plays that nicely with pint

**Comment by monroews** [comment]
The test for this code is incorrect and that is why it is failing. I suggest using the output from this code to generate a new test case.

**Review by eak24 — COMMENTED** [review]
As long as the tests are updated, this looks fine. The tests are here: https://github.com/AguaClara/aguaclara/blob/master/tests/design/test_lfom.py and you'll need to update them with the new expected values and push them to this branch.

**Comment by oliver-leung** [comment]
It looks like the Travis build is failing due to an error in the `width_stout` function.

```python
def width_stout(self, z):
    """Return the width of a Stout weir at elevation z. More info
    here. <https://confluence.cornell.edu/display/AGUACLARA/
    LFOM+sutro+weir+research>
    """
    w_per_flow = 2 / ((2 * pc.gravity * z) ** (1 / 2) *
                      con.VC_ORIFICE_RATIO * np.pi * self.hl)
    return w_per_flow
```

`pytest` claims that the return value of the function is seconds / meters ^2, and the code does look like it would return that. However, our test requires the return to be in meters. Given that the variable is called `w_per_flow`, should we be multiplying this quantity by our flow rate (which would give meters)?

@monroews

**Comment by monroews** [comment]
I believe the units are correct. m per m^3/s gives s/m^2. I think the test
case is incorrect.

Monroe Weber-Shirk | Senior Lecturer

Phone: +1 (607) 216-8445  | mw24@cornell.edu



AguaClara Cornell | Safe Water on Tap

Cornell University | 265 Hollister Hall | Ithaca, NY 14853 |
AguaClara.cornell.edu <http://aguaclara.cee.cornell.edu/>

<http://aguaclara.cornell.edu/>

**Comment by oliver-leung** [comment]
I'm still a bit skeptical - shouldn't a function with "width" in its name be returning meters? I feel that either the function's name or body should be changed to more accurately reflect what it does.

**Comment by monroews** [comment]
okay. Change the name to width per flow to match the name used within the function.

This function is being used consistently with the way I've defined it. So just change the name.

**Review by oliver-leung — APPROVED** [review]

**Pull request merged**